cylc/flow/data_store_mgr.py

```diff
@@ -2089,12 +2097,38 @@ def _family_ascent_point_update(self, fp_id):
                     is_queued_total += 1
 
                 tp_runahead = tp_delta
-                if (tp_runahead is None
-                        or not tp_runahead.HasField('is_runahead')):
+                if (
```

Comment:
Slightly off topic (happy to create my own refactor PR since I've the branch), but there's a lot of repetition here:

```python
    @staticmethod
    def _helper(tp_delta, tp_node, label):
        this = tp_delta
        if this is None or not this.HasField(label):
            this = tp_node
        value = getattr(this, label, None)
        if value:
            return value
        return None
```